### DISCUSSION TITLE
Which cache can be helpful for `z_fr_iss` ?

### DISCUSSION THREAD

**Discussion by galaxy001** [openzfs/zfs#12875] [Q&A]
I am using ZFS on a HDD partition with compress and dedup, after delete large files, `iotop` shows `z_fr_iss` takes much I/O and the HDD is too busy to write other stuff in.

I am plan to add some cache to solve this. There are `log`, `meta`, `cache` and `special`, which will be helpful ?

**Comment by FredFirestone** [comment]
Since you've got no answers so far at all here are my thoughts:

First of all: I assume it is not related to issue #3976 (too many tasks for cpu) / you are using a current version where spa.c has evolved to regard cpu, right?

Then you can have a look at special allocation classes (#3779) or a dedup vdev (https://openzfs.github.io/openzfs-docs/man/7/zpoolconcepts.7.html), but I'd assume you have to recreate your pool in order to utilize either one (for speeding up deletion of already stored data).
I doubt a cache device will be of much good as a) it has to be loaded to be of any use and b) DDT has to be modified anyway requiring writes on the device holding DDT being your bottleneck in the first place.
